Microwaves now perform better with stronger parts + general microwave code cleanup

Ilysen opened the pull request:
:cl: Ilysen
rscadd: Added beeping to microwaves!
rscadd: You can now use botanical satchels on microwaves to empty out any valid ingredients it has into the microwave.
tweak: Microwaves are now speedier, more efficient, and less likely to break when upgraded with better parts.
tweak: You can use anything with space cleaner contained in it to clean microwaves, instead of just spray bottles.
tweak: Adjusted messages and feedback for microwaves for clarity and consistency.
/:cl:

Microwaves still use the very ancient forms of microwave code, and are riddled with `src.`, bad documentation, and using `0` and `1` for false and true. In addition, even though microwaves can accept better parts, they don't interact with them in any way. I wanted to fix those!

### Technical Changes
* Removed all `src.` calls.
* Replaced applicable instances of `0` and `1` with `FALSE` and `TRUE`, when the relevant variable was only used to track bools.
* Renamed the `dirty` variable to `dirtiness`, as it's tracked as a scaling value and not a bool.
* Adjusted styling in `microwave.dm` for readability and smooth code flow.
* Changed the messages that display in the microwave's UI when it's broken or clean to offer some input on why it isn't working.
* Changed a few messages displayed by the microwave to unify grammar tense and give more helpful feedback.

### Mechanical Changes
* Added a sassy description.
* The microwave now beeps when interacting with it and when cooking starts.
* Cooking speed now increases with higher-tier micro-lasers.
  * Two max-tier lasers cause cooking to take 4 seconds, as opposed to the 10 seconds of the stock microwave.
* Power efficiency while cooking now increases with higher micro manipulator tier.
  * A max-tier manipulator uses 25% of the power that a stock microwave uses. This only applies to power spent during cooking, and doesn't affect passive draw.
* Matter bins now reduce the chance of microwaves dirtying or breaking.
  * Stock matter bins provide no reduction.
  * A max-tier matter bin will prevent breaks in 66% of cases.
  * Cooking will still be interrupted and failed when a break would occur; this just has a chance to prevent needing to fix it.

### Rationale

The code changes are needful - microwave code as it stands right now is prehistoric, back in the days when cavemen scratched on walls and used `src.` when referencing their variables. Now we're slightly more advanced cavemen, and it's astute to replace old practices where possible to update them to modern standards.

Beeping is being introduced as a major character arc in microwave's story. It is downright sinful to have a microwave without beeping. Sinful, I say.

Part scaling was done because, although the microwave is readily willing to accept part upgrades, these upgraded parts don't actually affect the microwave in any way, making it a moot point. I felt that adding upgrade functions to the microwave gave a little oomph if you wanted to go out of your way to request a part upgrade, as well as dispelled the problem of what seemed to be an oversight, given that many, many other machines (including the microlathe right nearby!) greatly benefit from part upgrades.

I asked around a bit on the bay discord when querying how folks would like the part change, and got one 'yes' and three answers that didn't give a direct 'yes' or 'no', so I figured I'd make it and see how people felt about it. I can understand if people don't like the idea of part upgrades, however, and I'm willing to get rid of them and just keep the code changes if I'm asked to. The code needs the update badly.

Tested fairly extensively on a local server.

If there are any issues with this PR, let me know! I'm always open to feedback on discord and here.

Commit:
Microwave cleanup

Spookerton commented on code/game/machinery/kitchen/microwave.dm:
```diff
@@ -55,86 +77,106 @@
+		var/has_cleaner = O.reagents != null && O.reagents.has_reagent(/datum/reagent/space_cleaner, 5)
+
+		// If they're trying to clean it, let them
+		if (has_rag || has_cleaner)
```
This branch falls through after running and transfers from the cleaner container to the microwave after cleaning ; should return.

Ilysen replied:
Eugh! I even rewrote this block specifically to have a return in it, then probably forgot it when I touched up. Sorry!

Ilysen commented:
I implemented a return function for cleaning the microwave. While I was in the neighborhood, I also implemented interactions between the microwave and satchels (mainly the botany one, in this case), allowing you to offload ingredients from the satchel into the microwave by using the satchel on it.

Spookerton commented on code/game/machinery/kitchen/microwave.dm:
```diff
@@ -55,86 +77,132 @@
+				broken = 0 // just to be sure
+				update_icon()
+				atom_flags = ATOM_FLAG_NO_TEMP_CHANGE | ATOM_FLAG_OPEN_CONTAINER
+			return
```
`return TRUE` to prevent afterattack reagent transfer

Spookerton commented on code/game/machinery/kitchen/microwave.dm:
```diff
@@ -55,86 +77,138 @@
+
+	else if(istype(O, /obj/item/weapon/storage))
+		if (LAZYLEN(ingredients) >= SScuisine.microwave_maximum_item_storage)
+			to_chat(user, SPAN_WARNING("[src] is completely full!</span>"))
```
spare closing tag inside macros
->
`SPAN_WARNING("[src] is completely full!")`

Ilysen replied:
buh, I figured I missed something

Spookerton commented on code/game/machinery/kitchen/microwave.dm:
```diff
@@ -55,86 +77,138 @@
+		if (objects_loaded)
+			if (!P.contents.len)
+				user.visible_message(SPAN_NOTICE("\The [user] empties \the [P] into \the [src]."),
+				SPAN_NOTICE("You empty \the [P] into \the [src].</span>"))
```
->
`SPAN_NOTICE("You empty \the [P] into \the [src].")`

Spookerton commented on code/game/machinery/kitchen/microwave.dm:
```diff
@@ -55,86 +77,138 @@
+				SPAN_NOTICE("You empty \the [P] into \the [src].</span>"))
+			else
+				user.visible_message(SPAN_NOTICE("\The [user] empties \the [P] into \the [src]."),
+				SPAN_NOTICE("You empty what you can from \the [P] into \the [src].</span>"))
```
->
`SPAN_NOTICE("You empty what you can from \the [P] into \the [src].")`

Spookerton commented on code/game/machinery/kitchen/microwave.dm:
```diff
@@ -55,86 +77,138 @@
+			return TRUE
+			
+		else
+			to_chat(user, SPAN_WARNING("\The [P] doesn't contain any compatible items to put into \the [src]!</span>"))
```
->
`SPAN_WARNING("\The [P] doesn't contain any compatible items to put into \the [src]!")`

Spookerton reviewed (dismissed):
fingers crossed that's the last bits.

Spookerton commented:
@Ilysen Before I forget again, please squash this.

Ilysen commented:
Bit of a learning curve with my client, but I think that'll do it.

Ilysen commented:
Sorry for the notif - is there anything else I should patch up with this?
